src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -171,6 +171,9 @@ public TimeSpan FirstBroadcastAttemptTimeout
             }
         }
 
+        public event EventHandler<HttpRequestMessage> OnBeforeRequest;
+        public event EventHandler<HttpResponseMessage> OnAfterRequests;
```

Comment:
`BeforeRequest` and `AfterRequest`

The `On` prefix is for the invocation method, usually

Reply:
We have `OnFailedRequest` so may it worth for consistency?
In addition, maybe while we have `OnFailedRequest` the `OnAfterRequests` should be `SucceedRequest`?

Reply:
Good idea, yes

Reply:
So, for "On" prefix I will add a property `FailedRequest` to wrap the `OnFailedRequest` and I will remove the "On" prefix from the two new. Is it make sense?

Reply:
We already have `FailedRequest` that is obsolete
```
[Obsolete("Use OnFailedRequest instead")]
public event Action<string, Exception> FailedRequest
```

Reply:
All our events start from 'On' so this is fine, but we need to have 'BeforeRequestEventArgs' and 'AfterRequestEventArgs'

Reply:
What about adding the `HttpRequestMessage` to `FailedRequestEventArgs` for correlation too?